assembly/src/parsers/u32_ops.rs

```diff
@@ -152,37 +160,39 @@ pub fn parse_u32split(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), A
 /// inserted. Please refer to the docs of `handle_arithmetic_operation` for more details.
 ///
 /// VM cycles per mode:
-/// - u32add: 4 cycles
-/// - u32add.b: 5 cycles
-/// - u32add.full: 2 cycles
-/// - u32add.unsafe: 1 cycle
-pub fn parse_u32add(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    handle_arithmetic_operation(span_ops, op, Operation::U32add)
+/// - u32checked_add: 4 cycles
+/// - u32checked_add.b:
+///    - 6 cycles if b is 1
+///    - 5 cycles if b is not 1
+/// - u32wrapping_add: 2 cycles
+/// - u32wrapping_add.b: 3 cycles
+/// - u32overflowing_add: 1 cycles
+/// - u32overflowing_add.b: 2 cycles
+pub fn parse_u32add(
+    span_ops: &mut Vec<Operation>,
+    op: &Token,
+    op_type: Type,
+) -> Result<(), AssemblyError> {
+    handle_arithmetic_operation(span_ops, op, Operation::U32add, op_type)
 }
 
-/// Translates u32add3 assembly instruction to VM operations.
+/// Translates u32unchecked_add3 assembly instruction directly to `U32ADD3` operation.
 ///
-/// In the unsafe mode this translates directly to `U32ADD3` operation. In the safe mode,
-/// we also assert that all three inputs are u32 values.
-///
-/// VM cycles per mode:
-/// - u32add3: 4 cycles
-/// - u32add3.unsafe: 1 cycle
-pub fn parse_u32add3(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+/// This operation takes 1 VM cycle.
+pub fn parse_u32unchecked_add3(
+    span_ops: &mut Vec<Operation>,
+    op: &Token,
+    op_type: Type,
```

Comment:
Shouldn't this be called just `parse_u32add3()` and the mode info would be passed via the parameter?